### PR TITLE
feat: Client code generated from OpenAPI spec

### LINKED ISSUE #63: feat: Code generated client from OpenAPI spec
**Is your feature request related to a problem? Please describe.**
As a 3rd party package we react to changes made by mistral. It's hard to keep track of changes and do them in timely manner.

In the newest versions of official mistral clients code generation is used and we should probably follow the same thing.
Another problem is that [mistralai/client-js](https://github.com/mistralai/client-js) didn't receive the update - and we were using it as our base for changes.

**Describe the solution you'd like**
Generate most of the client code from OpenAPI spec to allow faster updates of the package.

**Describe alternatives you've considered**
- Stay with the manually crafted code - okish but will require more time
- Leave package as it is - no further changes to the package - deprecation - not the option atm

**Additional context**
We want to keep package up to date as long as it is possible without too much effort on our side as maintenance takes time and we have not many resources to keep it going.

### PR DESCRIPTION
**Pull request by gawi151**
### Description

Generated client code + few scripts allowing us to create automation workflows

### Issue

Closes #63 

### Changes Made

- add generated client code (lib/src/generated directory)
- update examples to use new client code
- add script to detect openapi spec changes
- add script to generate client code from openapi spec
- add original and modified version of openapi spec that is compatible with code generator

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/nomtek/mistralai_client_dart/blob/main/CONTRIBUTING.md) document.
- [x] I have tested my changes.
- [x] I have added/updated documentation accordingly.

### CODE REVIEW

File: example/mistralai_client_dart_example.dart [hunk 1]
```diff
@@ -2,40 +2,41 @@
...
   // list models
   final modelsResult = await client.listModels();
-  final models = modelsResult.data.map((e) => e.id).toList();
-  print(models.join(', '));
+  final models = modelsResult.data?.map((e) => e.id).toList();
+  print(models?.join(', '));
 
   // chat without streaming
-  final params = ChatParams(
+  const params = ChatCompletionRequest(
```
Comment (KacperKluka) [inline example/mistralai_client_dart_example.dart]: Shouldn't it be called `request` instead of `params`? 🤔

**Review by KacperKluka — COMMENTED** [review]
Overall looks good 👍

File: example/mistralai_client_dart_example.dart (hunk 1 above)
Comment (gawi151) [reply example/mistralai_client_dart_example.dart]: Haha yeah probably would be best to be consistent with the class name. Thx for catching it.

Comment (gawi151) [reply example/mistralai_client_dart_example.dart]: Fixed in [e1c5263](https://github.com/nomtek/mistralai_client_dart/pull/69/commits/e1c5263ce4f704ab1e5f96036283eaefab4cdd1e)

**Review by KacperKluka — APPROVED** [review]
LGTM 👍 A lot of work, nicely done 💪

**Pull request merged**